### PR TITLE
HSEARCH-3559 Fix obsolete hibernate.search.indexes.<index name> syntax in the documentation

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3559

### CODE REVIEW

**Comment by gsmet** [comment]
Did you document how to affect an entity to a given backend?

**Comment by yrodiere** [comment]
> Did you document how to affect an entity to a given backend?

I had not, since `@Indexed` was not documented.

Added two commits to document `@Indexed`.

File: documentation/src/main/asciidoc/mapper-orm-mapping.asciidoc [hunk 1]
```diff
@@ -29,6 +29,72 @@ you can use a programmatic API to define the mapping.
...
+.Explicit index name with `@Indexed.index`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/search/documentation/mapper/orm/entityindexmapping/Author.java[tags=indexed-explicitindexname]
+----
+====
+
+If you defined multiple backends, you can map entities to another backend than the default one.
+By setting `@Indexed(backend = "backend2")` you inform Elasticsearch that the index
```
Comment (gsmet) [inline documentation/src/main/asciidoc/mapper-orm-mapping.asciidoc]: s/Elasticsearch/Hibernate Search/ ?

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**